### ISSUE TITLE
Allow building using system libappimage

### ISSUE THREAD

**Issue by jayvdb** [AppImage/AppImageKit#1165]
Many distros have libappimage https://repology.org/project/libappimage/versions

They expect that an AppImageKit build on their build farms will use their existing libappimage package, similar to how `USE_SYSTEM_MKSQUASHFS` works here.

Obviously they are then responsible for making sure their libappimage and other deps are appropriately patched until all relevant patches are upstreamed , i.e. https://github.com/AppImage/AppImageKit/issues/815

**Comment by lalten** [comment]
This would be interesting to me! Currently the repo can not be built without checking out the git submodules.
Previously I tried to build this as a third-party dependency in Bazel using http_archive (Which downloads the code archives automatically generated by GitHub - which [do not include submodules](https://github.com/dear-github/dear-github/issues/214)) and [rules_foreign_cc / CMake](https://bazelbuild.github.io/rules_foreign_cc/0.4.0/cmake.html). This fails because the archive does not contain the CMakeLists.txt being included to build libappimage.
If `USE_SYSTEM_MKSQUASHFS` was implemented I could first build libappimage separately and then provide it to the build for appimagetool :+1:
